src/Microsoft.JSInterop/DotNetDispatcher.cs

```diff
@@ -101,16 +126,26 @@ private static object InvokeSynchronously(string assemblyName, string methodIden
             }
 
             // Second, convert each supplied value to the type expected by the method
-            var serializerStrategy = SimpleJson.SimpleJson.CurrentJsonSerializerStrategy;
+            var runtime = (JSRuntimeBase)JSRuntime.Current;
+            var serializerStrategy = runtime.ArgSerializerStrategy;
             for (var i = 0; i < suppliedArgsLength; i++)
             {
-                suppliedArgs[i] = serializerStrategy.DeserializeObject(
-                    suppliedArgs[i], parameterTypes[i]);
+                if (parameterTypes[i] == typeof(JSAsyncCallResult))
+                {
+                    // For JS async call results, we have to defer the deserialization until
+                    // later when we know what type it's meant to be deserialized as
+                    suppliedArgs[i] = new JSAsyncCallResult(suppliedArgs[i]);
```

Comment:
I'm not totally sure I grok what's going on with this type. How is this different from using `object` ? What are the scenarios where you'd use this? Do we expect library code to need this?

Reply:
A fair question - it was definitely not made clear in the code. Hopefuly I've now clarified this in 233c392, but please let me know if you think that still doesn't suffice.